examples/structured_data/tabtransformer.py

```diff
@@ -0,0 +1,537 @@
+test_data.income_bracket = test_data.income_bracket.apply(
+    lambda value: value.replace(".", "")
+)
+
+"""
+Now we store the training and test data in separate CSV files.
+"""
+
+train_data_file = "train_data.csv"
+test_data_file = "test_data.csv"
+
+train_data.to_csv(train_data_file, index=False, header=False)
+test_data.to_csv(test_data_file, index=False, header=False)
+
+"""
+## Define dataset metadata
+
+Here, we define the metadata of the dataset that will be useful for reading and parsing
+the data into input features, and encoding the input features with respect to their types.
+"""
+
+# A list of the numerical feature names.
+NUMERIC_FEATURE_NAMES = [
+    "age",
+    "education_num",
+    "capital_gain",
+    "capital_loss",
+    "hours_per_week",
+]
+# A dictionary of the categorical features and their vocabulary.
+CATEGORICAL_FEATURES_WITH_VOCABULARY = {
+    "workclass": sorted(list(train_data["workclass"].unique())),
+    "education": sorted(list(train_data["education"].unique())),
+    "marital_status": sorted(list(train_data["marital_status"].unique())),
+    "occupation": sorted(list(train_data["occupation"].unique())),
+    "relationship": sorted(list(train_data["relationship"].unique())),
+    "race": sorted(list(train_data["race"].unique())),
+    "gender": sorted(list(train_data["gender"].unique())),
+    "native_country": sorted(list(train_data["native_country"].unique())),
+}
+# Name of the column to be used as instances weight.
+WEIGHT_COLUMN_NAME = "fnlwgt"
+# A list of the categorical feature names.
+CATEGORICAL_FEATURE_NAMES = list(CATEGORICAL_FEATURES_WITH_VOCABULARY.keys())
+# A list of all the input features.
+FEATURE_NAMES = NUMERIC_FEATURE_NAMES + CATEGORICAL_FEATURE_NAMES
+# A list of column default values for each feature.
+COLUMN_DEFAULTS = [
+    [0.0] if feature_name in NUMERIC_FEATURE_NAMES + [WEIGHT_COLUMN_NAME] else ["NA"]
+    for feature_name in CSV_HEADER
+]
+# The name of the target feature.
+TARGET_FEATURE_NAME = "income_bracket"
+# A list of the labels of the target features.
+TARGET_LABELS = [" <=50K", " >50K"]
+
+"""
+## Configure the hyperparameters
+
+The hyperparameters includes model architecture and training configurations.
+"""
+
+learning_rate = 0.001
+weight_decay = 0.0001
+dropout_rate = 0.2
+batch_size = 265
+num_epochs = 15
+
+num_transformer_blocks = 3  # Number of transformer blocks.
+num_heads = 4  # Number of attention heads.
+embedding_dims = 16  # Embedding dimensions of the categorical features.
+mlp_hidden_units_factors = [
+    2,
+    1,
+]  # MLP hidden layer units, as factors of the number of inputs.
+num_mlp_blocks = 2  # Number of MLP blocks in the baseline model.
+
+"""
+## Implement data reading pipeline
+
+We define an input function that reads and parses the file, then converts features
+and labels into a[`tf.data.Dataset`](https://www.tensorflow.org/guide/datasets)
+for training or evaluation.
+"""
+
+from tensorflow.keras.layers import StringLookup
```

Comment:
You can just access it via `layers`, no need to import the symbol itself

Reply:
Done